app/modules/node-app-service/main.tf

```diff
@@ -30,6 +30,16 @@ resource "azurerm_app_service" "app_service" {
     always_on     = "true"
     ftps_state    = "FtpsOnly"
     http2_enabled = true
+
+    dynamic "ip_restriction" {
+      for_each = var.inbound_vnet_connectivity == false ? [1] : []
```

Comment:
Nitpicky thing, but I usually follow a for_each or a count with a new line. Find it helps with readability.